conda-store-server/conda_store_server/__init__.py

```diff
@@ -46,10 +49,28 @@ def _version2_fmt(build: "Build") -> str:  # noqa: F821
         name = build.specification.name
         return f"{hash}-{timestamp}-{id}-{name}"
 
+    def _version3_fmt(build: "Build") -> str:  # noqa: F821
+        # Adds namespace here to separate builds performed by different users
+        # since all builds are stored in the same directory in v3, see
+        # Build.build_path in orm.py. Additionally, this also hashes the
+        # timestamp and build id just to make collisions very unlikely
+        namespace_name = build.environment.namespace.name
+        specification_hash = build.specification.sha256
```

Comment:
Can you remind how is this sha256 computed? From the input textarea, or from the output lockfile?

Reply:
See this code and where it's used in the codebase:

```py
def datastructure_hash(v):
    json_blob = json.dumps(recursive_sort(v))
    return hashlib.sha256(json_blob.encode("utf-8")).hexdigest()
```

It doesn't use the lockfile, it uses the specification (the user-provided text input).

Reply:
Ah sorry, just saw above.